repositories/org_repository.go

```diff
@@ -0,0 +1,68 @@
+package repositories
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/hierarchical-namespaces/api/v1alpha2"
+)
+
+const (
+	OrgNameLabel = "cloudfoundry.org/org-name"
+)
+
+type OrgRecord struct {
+	Name      string
+	GUID      string
+	Suspended bool
+	CreatedAt time.Time
+	UpdatedAt time.Time
+}
+
+type OrgRepo struct {
+	rootNamespace    string
+	privilegedClient client.Client
+}
+
+func NewOrgRepo(rootNamespace string, privilegedClient client.Client) *OrgRepo {
+	return &OrgRepo{
+		rootNamespace:    rootNamespace,
+		privilegedClient: privilegedClient,
+	}
+}
+
+func (r *OrgRepo) FetchOrgs(ctx context.Context, names []string) ([]OrgRecord, error) {
+	subnamespaceAnchorList := &v1alpha2.SubnamespaceAnchorList{}
+
+	options := []client.ListOption{client.InNamespace(r.rootNamespace)}
```

Comment:
Just want to confirm my understanding here...

A CF Org is 1-to-1 with an HNC `SubnamespaceAnchors` and given the event HNC is being used for something else in the cluster, the idea is that we'll have a dedicated CF root namespace so this will only return the correct `SubnamespaceAnchors` (for the CF we're targeting) right?

Reply:
Yep. Top level namespace, `cf` say, contains `SubnamespaceAnchors` for all CF Orgs in a 1:1 relationship. The HNC controller creates namespaces for these.

Similarly, `SubnamespaceAnchors` created in the namespaces corresponding to Orgs have a 1:1 relationship to CF Spaces. And again the HNC controller creates a namespace for these. These become the workload namespaces.

Reply:
Nice. I understand better now, thank you!